### DISCUSSION TITLE
Access Decimal class in frontend

### DISCUSSION THREAD

**Discussion by arashi-dev** [prisma/prisma#17272] [Q&A] [answered]
Hello. I am using trpc, nextjs and prisma

I have a `zod` schema:
```ts
export const decimalSchema = z
  .custom<Prisma.Decimal.Value>(
    (v) =>
      typeof v === "string" || typeof v === "number" || Prisma.Decimal.isDecimal(v)
  )
  .transform((v) => new Prisma.Decimal(v));
```

this schema is used to validate and transform input/output of a decimal value

I can use it in backend (trpc) but not in frontend. because I should not export `Prisma` class from `@prisma/client` to frontend.
I even tried importing Decimal from `import { Decimal } from "@prisma/client/runtime";` but it gives me `module 'fs' not found` error

how should I handle this?

**Comment by pantharshit00** [comment, marked as answer]
Can you provide a complete minimal reproduction? Also,`Decimal` type from Prisma is reexport from [decimal.js](https://github.com/MikeMcl/decimal.js/) so you can also use that.

**Reply by arashi-dev** [reply]
Yup. I used `decimal.js` and it's working now :))
thanks for your response

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
